wpimath/src/main/java/edu/wpi/first/math/estimator/KalmanFilter.java

```diff
@@ -187,6 +196,15 @@ public double getXhat(int row) {
   @SuppressWarnings("ParameterName")
   public void predict(Matrix<Inputs, N1> u, double dtSeconds) {
     this.m_xHat = m_plant.calculateX(m_xHat, u, dtSeconds);
+
+    // Pₖ₊₁⁺ = (I−Kₖ₊₁C)Pₖ₊₁⁻(I−Kₖ₊₁C)ᵀ + Kₖ₊₁RKₖ₊₁ᵀ
+    // Use Joseph form for numerical stability
+    m_P =
+        Matrix.eye(m_states)
+            .minus(m_K.times(m_C))
+            .times(m_P)
+            .times(Matrix.eye(m_states).minus(m_K.times(m_C)).transpose())
+            .plus(m_K.times(m_discR).times(m_K.transpose()));
```

Comment:
This equation should be in correct() instead. The predict() equation should be APAᵀ + Q.

Reply:
Sorry about that. I have moved the equation to the correct step.

I also removed the local C matrix from the correct step as there was already a member variable that grabbed the same matrix from the plant in the constructor.